lib/spack/spack/util/compression.py

```diff
@@ -186,47 +182,47 @@ def _lzma_decomp(archive_file):
                 shutil.copyfileobj(lar, ar)
+        return archive_out
     else:
-        if sys.platform == "win32":
-            return _7zip(archive_file)
-        else:
-            return _xz(archive_file)
+        return _xz(archive_file)
 
 
-def _win_compressed_tarball_handler(archive_file):
+def _win_compressed_tarball_handler(helper):
     """Decompress and extract compressed tarballs on Windows.
-    This method uses 7zip in conjunction with the tar utility
-    to perform decompression and extraction in a two step process
-    first using 7zip to decompress, and tar to extract.
-
-    The motivation for this method is the inability of 7zip
-    to directly decompress and extract compressed archives
-    in a single shot without undocumented workarounds, and
-    the Windows tar utility's lack of access to the xz tool (unsupported on Windows)
+    This method uses a decompression method in conjunction with
+    the tar utility to perform decompression and extraction in
+    a two step process first using decompressor to decompress,
+    and tar to extract.
+
+    The motivation for this method is Windows tar utility's lack
+    of access to the xz tool (unsupported natively on Windows) but
+    can be installed manually or via spack
     """
-    # perform intermediate extraction step
-    # record name of new archive so we can extract
-    # and later clean up
-    decomped_tarball = _7zip(archive_file)
-    # 7zip is able to one shot extract compressed archives
-    # that have been named .txz. If that is the case, there will
-    # be no intermediate archvie to extract.
-    if check_extension(decomped_tarball, "tar"):
-        # run tar on newly decomped archive
-        outfile = _untar(decomped_tarball)
-        # clean intermediate archive to mimic end result
-        # produced by one shot decomp/extraction
-        os.remove(decomped_tarball)
-        return outfile
-    return decomped_tarball
+
+    def unarchive(archive_file):
+        # perform intermediate extraction step
+        # record name of new archive so we can extract
+        # and later clean up
+        decomped_tarball = helper(archive_file)
+        # 7zip is able to one shot extract compressed archives
+        # that have been named .txz. If that is the case, there will
+        # be no intermediate archvie to extract.
+        if check_extension(decomped_tarball, "tar"):
+            # run tar on newly decomped archive
+            outfile = _untar(decomped_tarball)
+            # clean intermediate archive to mimic end result
+            # produced by one shot decomp/extraction
+            os.remove(decomped_tarball)
+            return outfile
+        return decomped_tarball
+
+    return unarchive
 
 
 def _xz(archive_file):
     """Decompress lzma compressed .xz files via xz command line
     tool. Available only on Unix
     """
-    if sys.platform == "win32":
-        raise RuntimeError("XZ tool unavailable on Windows")
```

Comment:
The docstring should be updated to reflect this change. The PR initial comment doesn't mention this but is `xz` availability a recent change on Windows?

Reply:
Yes it should, nice catch. XZ is still not available natively on Windows, however, this is more rectifying an inconsistency. Tar is the only tool in this compression module that is available natively on Windows. For unzip we shell out to tar, and for gz, bz2, and others if we can't use the python module for one reason or another we try to find something on the system and fail if nothing is found. This PR refactors the xz support to be the same as the others. Previously the xz python module wasn't even an option prior to Python3.4 (3.3?) so we had to have a fallback/default to 7zip as we just needed some way to extract Clingo. Now that we can more consistently (on Windows it should always be present) rely on the xz python module, we can refactor this to have the same fallback scheme as the rest. The end goal being not having a dependency on 7zip when spinning up Spack for the first time.